### PR TITLE
CLOUDSTACK-9238: Fix URL length to 2048 for all url fields in VO

### PR DESCRIPTION
**Pull request by marcaurele**
I will update the PR to add max field length in the API commands too

### CODE REVIEW

**Comment by karuturi** [comment]
Why dont we change the default length in [﻿GenericDaoBase](https://github.com/apache/cloudstack/blob/master/framework/db/src/com/cloud/utils/db/GenericDaoBase.java#L1510) and [api.Parameter](https://github.com/apache/cloudstack/blob/master/api/src/org/apache/cloudstack/api/Parameter.java#L47) ?
If possible, get it from some global constant

File: setup/db/db/schema-471to480.sql [hunk 1]
```diff
@@ -27,3 +27,4 @@ ALTER TABLE `cloud`.`image_store` MODIFY COLUMN `url` varchar(2048);
 ALTER TABLE `cloud`.`template_store_ref` MODIFY COLUMN `url` varchar(2048);
 ALTER TABLE `cloud`.`volume_store_ref` MODIFY COLUMN `url` varchar(2048);
 ALTER TABLE `cloud`.`volume_store_ref` MODIFY COLUMN `download_url` varchar(2048);
+ALTER TABLE `cloud`.`upload` MODIFY COLUMN `url` varchar(2048);
```
Comment (karuturi) [inline setup/db/db/schema-471to480.sql]: I think this table isnt used anymore.

**Comment by marcaurele** [comment]
Because not all varchar columns have been updated to a length of 2048 in the DB, so there will be a mismatche between API, VO and the database.

**Comment by karuturi** [comment]
ahh.. my bad.. thanks for the clarification.
so, we need to introduce a new CommandType.URL for api parameters and use java.net.URL in VOs.

LGTM in the current state as well as this fixes the issue. 👍

**Comment by footplus** [comment]
Seems to break the registerIso command here... Will check exactly why.

``
2016-05-27 15:01:14,211 ERROR [c.c.a.ApiServer] (catalina-exec-1:ctx-56423a5c ctx-9b38033e) (logid:68ac142d) unhandled exception executing api command: [Ljava.lang.String;@475199b7
java.lang.NullPointerException
	at com.cloud.template.TemplateAdapterBase.prepare(TemplateAdapterBase.java:149)
	at com.cloud.template.TemplateAdapterBase.prepare(TemplateAdapterBase.java:317)
	at com.cloud.template.HypervisorTemplateAdapter.prepare(HypervisorTemplateAdapter.java:124)
	at com.cloud.template.TemplateManagerImpl.registerIso(TemplateManagerImpl.java:299)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:606)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:317)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:183)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:150)
	at org.apache.cloudstack.network.contrail.management.EventUtils$EventInterceptor.invoke(EventUtils.java:106)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:161)
	at com.cloud.event.ActionEventInterceptor.invoke(ActionEventInterceptor.java:51)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:161)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:91)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:172)
	at org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:204)
	at com.sun.proxy.$Proxy179.registerIso(Unknown Source)
	at org.apache.cloudstack.api.command.user.iso.RegisterIsoCmd.execute(RegisterIsoCmd.java:192)
	at com.cloud.api.ApiDispatcher.dispatch(ApiDispatcher.java:150)
	at com.cloud.api.ApiServer.queueCommand(ApiServer.java:698)
	at com.cloud.api.ApiServer.handleRequest(ApiServer.java:529)
	at com.cloud.api.ApiServlet.processRequestInContext(ApiServlet.java:297)
	at com.cloud.api.ApiServlet$1.run(ApiServlet.java:127)
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext$1.call(DefaultManagedContext.java:56)
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.callWithContext(DefaultManagedContext.java:103)
	at org.apache.cloudstack.managed.context.impl.DefaultManagedContext.runWithContext(DefaultManagedContext.java:53)
	at com.cloud.api.ApiServlet.processRequest(ApiServlet.java:124)
	at com.cloud.api.ApiServlet.doGet(ApiServlet.java:86)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:617)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:723)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:290)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:206)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:233)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:191)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:127)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:102)
	at org.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:555)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:109)
	at org.apache.catalina.valves.RemoteIpValve.invoke(RemoteIpValve.java:600)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:299)
	at org.apache.coyote.http11.Http11NioProcessor.process(Http11NioProcessor.java:889)
	at org.apache.coyote.http11.Http11NioProtocol$Http11ConnectionHandler.process(Http11NioProtocol.java:732)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.run(NioEndpoint.java:2274)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
	at java.lang.Thread.run(Thread.java:745)
``

**Comment by swill** [comment]
Ok, this is building in my CI environment right now, so I will see if it fails for me as well.  Thx...

**Comment by swill** [comment]
### CI RESULTS

```
Tests Run: 85
  Skipped: 0
   Failed: 0
   Errors: 0
 Duration: 3h 56m 33s
```



**Associated Uploads**

**`/tmp/MarvinLogs/DeployDataCenter__May_27_2016_15_27_02_O0ILJL:`**
* [dc_entries.obj](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/DeployDataCenter__May_27_2016_15_27_02_O0ILJL/dc_entries.obj)
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/DeployDataCenter__May_27_2016_15_27_02_O0ILJL/failed_plus_exceptions.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/DeployDataCenter__May_27_2016_15_27_02_O0ILJL/runinfo.txt)

**`/tmp/MarvinLogs/test_network_VEWE3L:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/test_network_VEWE3L/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/test_network_VEWE3L/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/test_network_VEWE3L/runinfo.txt)

**`/tmp/MarvinLogs/test_vpc_routers_OY7CL0:`**
* [failed_plus_exceptions.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/test_vpc_routers_OY7CL0/failed_plus_exceptions.txt)
* [results.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/test_vpc_routers_OY7CL0/results.txt)
* [runinfo.txt](https://objects-east.cloud.ca/v1/e465abe2f9ae4478b9fff416eab61bd9/PR1567/tmp/MarvinLogs/test_vpc_routers_OY7CL0/runinfo.txt)


Uploads will be available until `2016-07-27 02:00:00 +0200 CEST`

*Comment created by [`upr comment`](https://github.com/cloudops/upr).*

**Comment by swill** [comment]
This is coming back clean.  I will merge it...

**Pull request merged**